recipes-bsp/u-boot/u-boot-iot2050.inc

```diff
@@ -64,6 +74,13 @@ dpkg_runbuild_prepend() {
     export TEE=/usr/lib/optee-os/iot2050/tee-pager_v2.bin
 }
 
+dpkg_runbuild_append_secureboot() {
+    if [ "${IOT2050_OTPCMD}" = "1" ]; then
+        dd if=${BUILDCHROOT_DIR}/usr/lib/secure-boot-otp-provisioning/otpcmd.bin \
+            of=${S}/${SPI_FLASH_IMG} bs=64K seek=108 count=1 conv=notrunc
```

Comment:
Why do we need `count=1` here? Is dd doing any padding this way? Do we need padding?

Also, this would be easier verifiable against the layout:
`dd bs=$((0x10000) seek=$((0x6c)) ...`

Reply:
Will check later again if we cannot also use binman for that, either post build (`binman replace`) or by making the otpcmd payload optional via kconfig.

Reply:
Will check later again if we cannot also use binman for that, either post build (`binman replace`) or by making the otpcmd payload optional via kconfig.

Reply:
We can drop count=1 here, the otpcmd.bin is nothing but x509 + FIT image, both are size-deterministic, padding is irrelevant.

Will check if binman could do this trick.

Reply:
Just noticed in the `iot2050-sign-fw.sh`:

```
# currently broken in upstream
#source/tools/binman/binman replace -i flash.bin -f tispl.bin_signed blob@0x180000
dd if=tispl.bin_signed of=flash.bin bs=$((0x1000)) seek=$((0x180000/0x1000)) conv=notrunc
```

Reply:
Seems bitbake could not handle `$(())`:

```
ERROR: /repo/recipes-bsp/u-boot/u-boot-iot2050-pg1_2022.01.bb: Error during parse shell code, the last 5 lines are:                                                            | ETA:  0:00:00
    if [ "0" = "1" ]; then
        dd if=/build/tmp/deploy/buildchroot-host/debian-bullseye-amd64_iot2050-debian-arm64/usr/lib/secure-boot-otp-provisioning/otpcmd.bin \
            of=/build/tmp/work/iot2050-debian-arm64/u-boot-iot2050-pg1/2022.01-r0/u-boot-2022.01/flash.bin bs=$((0x10000)) seek=$((0x6c)) conv=notrunc
    fi

WARNING: /repo/recipes-bsp/u-boot/u-boot-iot2050-pg1_2022.01.bb: Exception during build_dependencies for dpkg_runbuild
WARNING: /repo/recipes-bsp/u-boot/u-boot-iot2050-pg1_2022.01.bb: Error during finalise of /repo/recipes-bsp/u-boot/u-boot-iot2050-pg1_2022.01.bb
```

Reply:
Hmm...https://bugzilla.yoctoproject.org/show_bug.cgi?id=11314

Reply:
Too bad... Let me look into adding an optional binary to k3-am65-iot2050-boot.dtsi again.

Reply:
Does [this](https://github.com/siemens/meta-iot2050/blob/a45a50adc4e92c0332d1380b0342d51826a30526/recipes-bsp/u-boot/files/0024-arm-dts-iot2050-Use-external-blob-file-for-OTP-updat.patch) do the trick?

Reply:
It's a building block. But it should be made configurable and optional (i.e. warning free). I'm on it.

Reply:
https://github.com/siemens/u-boot/commit/3474476dd2b3a4f7779f13e125bde983c0c4778a

Reply:
Merged and refact the otpcmd to `OVERRIDES .= ":otpcmd"`, just like the `secure-boot.yml`